tensorflow_addons/layers/tcn.py

```diff
@@ -0,0 +1,336 @@
+                        filters=self.filters,
+                        kernel_size=self.kernel_size,
+                        dilation_rate=self.dilation_rate,
+                        padding=self.padding,
+                        name=name,
+                        kernel_initializer=self.kernel_initializer)
+                    self.residual_layers.append(conv_layer)
+
+                if self.use_batch_norm:
+                    batch_norm_layer = tf.keras.layers.BatchNormalization()
+                    self.residual_layers.append(batch_norm_layer)
+
+                self.residual_layers.append(tf.keras.layers.Activation('relu'))
+                self.residual_layers.append(
+                    tf.keras.layers.SpatialDropout1D(
+                        rate=self.dropout_rate))
+
+            if not self.last_block:
+                # 1x1 conv to match the shapes (channel dimension).
+                name = 'conv1D_{}'.format(k + 1)
+                with tf.name_scope(name):
+                    self.shape_match_conv = tf.keras.layers.Conv1D(
+                        filters=self.filters,
+                        kernel_size=1,
+                        padding='same',
+                        name=name,
+                        kernel_initializer=self.kernel_initializer)
+
+            else:
+                self.shape_match_conv = tf.keras.layers.Lambda(
+                    lambda x: x, name='identity')
+
+            self.final_activation = tf.keras.layers.Activation(self.activation)
+
+            # this is done to force keras to add the layers in the list to
+            # self._layers
+            for layer in self.residual_layers:
+                self.__setattr__(layer.name, layer)
+
+    def build(self, input_shape):
+
+        # build residual layers
+        self.res_output_shape = input_shape
+        for layer in self.residual_layers:
+            layer.build(self.res_output_shape)
+            self.res_output_shape = layer.compute_output_shape(
+                self.res_output_shape)
+
+        # build shape matching convolutional layer
+        self.shape_match_conv.build(input_shape)
+        self.res_output_shape = self.shape_match_conv.compute_output_shape(
+            input_shape)
+
+        # build final activation layer
+        self.final_activation.build(self.res_output_shape)
+
+        super(ResidualBlock, self).build(input_shape)
+
+    def call(self, inputs, training=None):
+        """
+        Returns: A tuple where the first element is the residual model tensor,
+                 and the second is the skip connection tensor.
+        """
+        x = inputs
+        for layer in self.residual_layers:
+            if isinstance(layer, tf.keras.layers.SpatialDropout1D):
+                x = layer(x, training=training)
+            else:
+                x = layer(x)
+
+        x2 = self.shape_match_conv(inputs)
+        res_x = tf.keras.layers.add([x2, x])
```

Comment:
Any reason for not doing:
```
res_x = x2 + x
```

Reply:
Originally I thought doing layers.add is more readable than just the + sign but after reading more code I think you are correct that using + is more conventional. Will update, thanks @Squadrick !

Reply:
updated!